examples/social_media/persist/model.bal

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/persist as _;
+import ballerina/time;
+
+public enum Gender {
+    MALE,
+    FEMALE
+}
+
+public type User record{|
+    readonly string firstName;
+    readonly string lastName;
+    string email;
+    int age;
+    time:Date dateOfBirth;
+    Gender gender;
+    boolean isMarried;
+    string spouseName?;
+	Post[] post;
+|};
+
+public type Post record{|
+    readonly int id;
+    string title;
+    string content;
+    User user;
+	Comment[] comment;
```

Comment:
```suggestion
    User user;
    Comment[] comment;
```

Reply:
Fixed in https://github.com/ballerina-platform/persist-tools/pull/333/commits/32874f0238f447941f8a9e3ebfae4872414ca2bb